### PR TITLE
Peak detection + tests, Actions

### PR DESCRIPTION
**Pull request by tandej**
Also added normalizeAngleDegrees to compliment normalizeAngleRadians, and patched up directory structure slightly.

Peak detection tests are unfortunately not as _concrete_ as I think they could be, but it's ultimately a really qualitative thing that can be tuned anyway.

### CODE REVIEW

**Comment by tandej** [comment]
Looks like this in action:
image

File: src/main/java/frc/team997/lib/math/PeakDetectionFilter.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+     * <p>Generally, there's a balance to be made between window and stddev and mean influences. A
+     * large window will result in more phase lag, but can allow for influences of 1.0 for stddev
+     * and mean, as signals will be surrounded by much more data. However, a small window has little
+     * phase lag, but can be susceptible to high influences from peaks (they are, after, all,
+     * outliers by definition).
+     *
+     * <p>Special care should be taken when the series exhibits trends over time to balance
+     * influences so as not to completely erase said trend.
+     *
+     * @param window Looks back this many cycles to determine mean and standard deviation.
```
Comment (elliotnash) [inline src/main/java/frc/team997/lib/math/PeakDetectionFilter.java]: I would change the window param docs to something like The number of cycles used to calculate the mean and standard deviation, starting with a verb doesn’t make sense for a parameter imo.

File: src/main/java/frc/team997/lib/math/PeakDetectionFilter.java [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    }
+
+    /**
+     * Calculates whether a given value is a peak, valley, or neither relative to the rest of the
+     * series.
+     *
+     * @param value The double value to evaluate.
+     * @return 1 if the value is a peak, -1 if it is a valley, or 0 if it is neither.
+     */
+    public int calculate(double value) {
```
Comment (elliotnash) [inline src/main/java/frc/team997/lib/math/PeakDetectionFilter.java:125]: does this detect peaks if they’re window isn’t full? if not, i’d add that to the docs

File: src/test/java/frc/team997/lib/math/PeakDetectionFilterTests.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+import java.security.InvalidParameterException;
+import org.junit.Test;
+
+/**
+ * Tests for the PeakDetectionFilter.
+ *
+ * <p>Unfortunately, it's impractical to automate the kind of qualitative tests we want, so this
```
Comment (elliotnash) [inline src/test/java/frc/team997/lib/math/PeakDetectionFilterTests.java:29]: this is probably the best way to do this - however I would add tests for different edge cases of data (think, a dataset that doesn't fill up the window, a dataset that has no peaks, a dataset that has a prolonged/gradual peak, etc.

File: src/main/java/frc/team997/lib/math/PeakDetectionFilter.java (hunk 2 above)
Comment (elliotnash) [reply src/main/java/frc/team997/lib/math/PeakDetectionFilter.java:125]: also, this adds the value to the frame, right? maybe specify that in docs

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    container: wpilib/roborio-cross-ubuntu:2022-18.04
+    
+    steps:
+      - uses: actions/checkout@v2
+      
+      - name: Grant execute permission to gradle
+        run: chmod +x gradlew
+      
+      - name: Compile and test
+        run: ./gradlew build
```
Comment (elliotnash) [inline .github/workflows/main.yml]: I would make build and test a different step - one ./gradlew build -x test and one ./graldew test

File: src/test/java/frc/team997/lib/math/PeakDetectionFilterTests.java (hunk 3 above)
Comment (elliotnash) [reply src/test/java/frc/team997/lib/math/PeakDetectionFilterTests.java:29]: also, in real use, I imagine you're going to probably want to be able to initialize it with no data, and fill up the data as you call calculate

**Review by elliotnash — APPROVED** [review]
lgtm
